src/Chr.Avro.Binary/BinaryDeserializerBuilder.cs

```diff
@@ -1392,8 +1393,20 @@ public override IBinaryDeserializerBuildResult BuildDelegate(TypeResolution reso
                         ExceptionDispatchInfo.Capture(indirect.InnerException).Throw();
                     }
 
-                    expression = GenerateConversion(expression, target);
+                    var immutableDictionary = typeof(ImmutableDictionary<,>)
+                        .MakeGenericType(key, item);
 
+                    if (target.Equals(immutableDictionary))
```

Comment:
in the array case, we override `GenerateConversion`: https://github.com/ch-robinson/dotnet-avro/blob/development/src/Chr.Avro.Binary/BinaryDeserializerBuilder.cs#L468

for consistency, it’s probably best to do the same here